### ISSUE TITLE
"unable to open database file" in Travis CI

### ISSUE THREAD

**Issue by JDSeiler** [JDSeiler/alexandria-db#1]
When running the the server and NodeJS tests locally, all tests pass. However, when running the same build of the server and tests in Travis, book#getById() (the only function to be hooked up to Sqlite so far) fails with the error message "enable to open database file ./db/dummy.db".

Currently, I am not sure why this error is occurring, since the rusqlite documentation specifies that `Connection::open()` will create the database if it does not exist. If rusqlite _was_ creating a new database, the function would fail with an error message similar to "no such table: book" since it would be a blank database.

It may be that `cargo run &` needs to be performed as sudo so that the Rust process has sufficient permissions to open files? I'll try that first to see if it makes a difference.

**Comment by JDSeiler** [comment]
Attempting to run cargo as sudo using `"sudo cargo run &` results in `sudo: cargo: command not found` and the server not being started at all. This is obviously will not result in passing tests.

**Comment by JDSeiler** [comment]
I think I have identified the issue. The file path to the database specified in `common.rs` was a relative path: `./db_storage/dummy.db`. As it so happens, the `.` in that path refers to whatever directory you were in when `cargo run` is executed. So if you're in `/src/dirA/dirB` `.` refers to `dirB` while running `cargo run` in `/src` makes `.` refer to `src`. The reason it was failing in Travis but not locally is that Travis runs cargo in the project root, while I was executing it locally in the src directory. In fact, running cargo in a directory outside of src (locally) produced the same error seen in Travis. By changing the file path to be relative to the project root: `./src/db_storage/dummy.db`, the error was resolved. 

This will serve as a band-aid fix for now but I would like to come up with a better way to find a path to the database, perhaps by resolving an absolute path somehow. I am unfamiliar with Rust's path manipulation capabilities so I'll have to do some research.
